pkg/endpoint/bpf.go

```diff
@@ -254,53 +248,38 @@ func hashHeaderfile(hashWriter hash.Hash, filepath string) (hash.Hash, error) {
 	return hashWriter, nil
 }
 
-func (e *Endpoint) runInit(libdir, rundir, epdir, ifName, debug string) error {
-	args := []string{libdir, rundir, epdir, ifName, debug, e.StringID()}
-	prog := filepath.Join(libdir, "join_ep.sh")
-
-	scopedLog := e.getLogger()
-
-	ctx, cancel := context.WithTimeout(context.Background(), ExecTimeout)
```

Comment:
This context was created before doing any Command execution. Can you do the same before calling the CompileandLoad in [0]? This way you are reusing a single ctx for everything call made. Be aware you will need to check for context.DeadlineExceeded before executing any preceding "expensive" function call.

Reply:
Is there a link somewhere or something that explains why you need to check `context.DeadlineExceeded` before executing subsequent calls? Won't the call just return an error that we can pass back?

Reply:
Ah, apparently it won't. Okay. I need to fix this up across the PR.

Reply:
And done.